### DISCUSSION TITLE
[Doc]: purpose of narwhals project

### DISCUSSION THREAD

**Discussion by frosk1** [narwhals-dev/narwhals#1249] [Q&A] [answered]
### What type of report is this?

Improvement

### Please describe the issue.

Why is this library not meant for end users? 

### If you have a suggestion on how it should be, add it below.

Narwhals is building a layer on top of dataframe backends, trying to abstract away the dataframe backend from its front end. Since the front end is used by end users, this library is for end users. I see narwhals as a chance (if the project is using its momentum) to enable a one and only python API for dataframe backends, defining a standard for dataframe front end. Meaning an end user only needs to learn one front end which is narwhals and does not care about how polars or pandas API work

**Comment by MarcoGorelli** [comment, marked as answer]
thanks for your question!

Narwhals is built with dataframe maintainers in mind - this is the target audience

Most end users would likely require a more extensive API surface than what we provide

However, if you're and end-user and find Narwhals useful, then by all means, you're welcome to use it - I'd be very interested in hearing what you're using it for, if you'd be willing to share, thanks 🙏
